frontend/src/metabase/dashboard/actions/sharing.ts

```diff
@@ -28,49 +28,58 @@ export const setSharing = (isSharing: boolean) => (dispatch: Dispatch) => {
 
 export const UPDATE_ENABLE_EMBEDDING =
   "metabase/dashboard/UPDATE_ENABLE_EMBEDDING";
-export const updateEnableEmbedding = createAction(
-  UPDATE_ENABLE_EMBEDDING,
-  ({ id }: DashboardIdPayload, enable_embedding: boolean) =>
-    DashboardApi.update({
-      id,
-      enable_embedding,
-    }),
-);
+export const updateEnableEmbedding = createAsyncThunk<
+  Dashboard,
+  Pick<Dashboard, "id" | "enable_embedding">
+>(UPDATE_ENABLE_EMBEDDING, async ({ id, enable_embedding }) => {
+  const response = await DashboardApi.update({
+    id,
+    enable_embedding,
+  });
+
+  return response;
+});
 
 export const UPDATE_EMBEDDING_PARAMS =
   "metabase/dashboard/UPDATE_EMBEDDING_PARAMS";
-export const updateEmbeddingParams = createAction(
-  UPDATE_EMBEDDING_PARAMS,
-  ({ id }: DashboardIdPayload, embedding_params: EmbedOptions) =>
-    DashboardApi.update({ id, embedding_params }),
-);
+export const updateEmbeddingParams = createAsyncThunk<
+  Dashboard,
+  Pick<Dashboard, "id" | "embedding_params">
+>(UPDATE_EMBEDDING_PARAMS, async ({ id, embedding_params }) => {
+  const response = await DashboardApi.update({ id, embedding_params });
+
+  return response;
```

Comment:
nit: we could return immediately as before if we don't do anything with the response